### PR TITLE
feat(scripts): Add more checks to the check function

### PR DESCRIPTION
**Pull request by Henryws**
## Purpose

Currently the `check` function doesn't do much, so I added more checks that look for vital variables and alert the user if there is an issue

### CODE REVIEW

File: misc/scripts/install-local.sh [hunk 1]
```diff
@@ -55,6 +55,21 @@ function checks() {
 	if [[ -z "$hash" ]]; then
 		fancy_message warn "Package does not contain a hash"
 	fi
```
Comment (wizard-28) [inline misc/scripts/install-local.sh:57]: Could we remove "Package does not contain a hash" message for `-git` pacscripts?

File: misc/scripts/install-local.sh [hunk 2]
```diff
@@ -55,6 +55,21 @@ function checks() {
...
+		exit 1
+	fi
+	if [[ -z "$name" ]]; then
+		fancy_message error "Package does not contain name"
+		exit 1
+	fi
+	if [[ -z "$url" ]]; then
+		fancy_message error "Package does not contain URL"
+		exit 1
+	fi
```
Comment (wizard-28) [inline misc/scripts/install-local.sh:72]:
```suggestion
	fi
	if [[ -z "$maintainer" ]]; then
		fancy_message warn "Package does not have a maintainer"
		fancy_message warn "It maybe no longer maintained. Please be advised."
	fi
```

I think we could add a warning for missing `maintainer` fields. So if some non-`-git` pacscript gets no longer updated for quite some time. We could remove that field, and the users will be warned accordingly.

Or the maintainer themselves could remove the field if they no longer wish to maintain the script.

**Review by wizard-28 — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh (hunk 2 above)
Comment (Henryws) [reply misc/scripts/install-local.sh:72]: This could be one line

Comment (wizard-28) [reply misc/scripts/install-local.sh:72]: What do you suggest for the one liner?

Comment (wizard-28) [reply misc/scripts/install-local.sh:72]: Too much is cramped into a single line. I prefer 2 line approach.

File: misc/scripts/install-local.sh [hunk 3]
```diff
@@ -52,8 +52,30 @@ function checks() {
...
 	fi
 
+	if [[ -z "$name" ]]; then
+		fancy_message error "Package does not contain name"
+		exit 1
+	fi
 	if [[ -z "$hash" ]]; then
-		fancy_message warn "Package does not contain a hash"
+		# if the name doesn't end in -git, we put up the prompt
+		if ! [[ "$name" == "*-git" ]]; then
```
Comment (wizard-28) [inline misc/scripts/install-local.sh:61]:
```suggestion
		if [[ "$name" != "*-git" ]]; then
```

Simplify condition.

**Review by wizard-28 — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh [hunk 4]
```diff
@@ -44,17 +44,36 @@ function trap_ctrlc () {
...
 function checks() {
+
+	if [[ -z "$name" ]]; then
+		fancy_message error "Package does not contain name"
+		exit 1
+	fi
+	if [[ -z "$hash" && "$name" != "*-git" ]]; then
+		fancy_message warn "Package does not contain a hash"
+	fi
+	if [[ -z "$description" ]]; then
```
Comment (NarutoXY) [inline misc/scripts/install-local.sh:55]: `description` is not necessary tbh

File: misc/scripts/install-local.sh [hunk 5]
```diff
@@ -44,17 +44,36 @@ function trap_ctrlc () {
...
+	fi
+	if [[ -z "$version" ]]; then
+		fancy_message error "Package does not contain version"
+		exit 1
+	fi
+	if [[ -z "$url" ]]; then
+		fancy_message error "Package does not contain URL"
+		exit 1
+	fi
+	if [[ -z "$maintainer" ]]; then
```
Comment (NarutoXY) [inline misc/scripts/install-local.sh:66]: `maintainer` is also not the essential ones. We can show them as a warning

File: misc/scripts/install-local.sh (hunk 4 above)
Comment (NarutoXY) [reply misc/scripts/install-local.sh:55]: oh wait did not notice you were not exiting. `maintainer` can be resolved but `description` should not show any warning as description is just for package maintainer's sanity

**Review by NarutoXY — COMMENTED** [review]
All good. There is a suggestion tho

File: misc/scripts/install-local.sh [hunk 6]
```diff
@@ -44,17 +44,36 @@ function trap_ctrlc () {
 
 # run checks to verify script works
 function checks() {
+
+	if [[ -z "$name" ]]; then
+		fancy_message error "Package does not contain name"
+		exit 1
+	fi
+	if [[ -z "$hash" && "$name" != "*-git" ]]; then
```
Comment (wizard-28) [inline misc/scripts/install-local.sh:52]:
```suggestion
	if [[ -z "$hash" && "$name" != *-git ]]; then
```

Fix `*-git` glob

**Review by wizard-28 — CHANGES_REQUESTED** [review]

**Review by wizard-28 — APPROVED** [review]
Works in my short test.

**Pull request merged**